### ISSUE TITLE
GDAX/Coinbase Pro parses response incorrectly in respect to the order size/amount

### ISSUE THREAD

**Issue by ajoseps** [ccxt/ccxt#5826]
- OS: Fedora 29
- CCXT version: v1.18.1150
- Exchange: GDAX/Coinbase Pro
- Method: fetchClosedOrders

MARKET BUY orders on GDAX return the order amount in quote (with field `funds`), whereas a MARKET SELL returns the order amount in base (with field `size`). In both of these case, CCXT parses the amount using the fields `size` or `funds` depending on what exists and reports it as the order `amount`. This seems incorrect because the MARKET BUY will always have its `amount` in quote whereas MARKET SELL, LIMIT BUY, LIMIT SELL will have their `amount`s in base.

**MARKET BUY**
```
  {
    "id": "RETRACTED",
    "info": {
      "id": "RETRACTED",
      "product_id": "LTC-USD",
      "side": "buy",
      "funds": "{funds_num}",
      "specified_funds": "{specified_funds_num}",
      "type": "market",
      "post_only": false,
      "created_at": "RETRACTED",
      "done_at": "RETRACTED",
      "done_reason": "filled",
      "fill_fees": "RETRACTED",
      "filled_size": "RETRACTED",
      "executed_value": "{executed_value_num}",
      "status": "done",
      "settled": true
    },
    "timestamp": RETRACTED,
    "datetime": "RETRACTED",
    "lastTradeTimestamp": null,
    "status": "closed",
    "symbol": "LTC/USD",
    "type": "market",
    "side": "buy",
    "price": null,
    "cost": RETRACTED,
    "amount": RETRACTED,
    "filled": RETRACTED,
    "remaining": RETRACTED,
    "fee": {
      "cost": RETRACTED,
      "currency": null,
      "rate": null
    }
  },
```
**MARKET SELL**
```
  {
    "id": "RETRACTED",
    "info": {
      "id": "RETRACTED",
      "size": "{size_num}",
      "product_id": "ETH-USD",
      "side": "sell",
      "type": "market",
      "post_only": false,
      "created_at": "RETRACTED",
      "done_at": "RETRACTED",
      "done_reason": "filled",
      "fill_fees": "RETRACTED",
      "filled_size": "RETRACTED",
      "executed_value": "{executed_value_num}",
      "status": "done",
      "settled": true
    },
    "timestamp": RETRACTED,
    "datetime": "RETRACTED",
    "lastTradeTimestamp": null,
    "status": "closed",
    "symbol": "ETH/USD",
    "type": "market",
    "side": "sell",
    "price": null,
    "cost": RETRACTED,
    "amount": RETRACTED,
    "filled": RETRACTED,
    "remaining": RETRACTED,
    "fee": {
      "cost": RETRACTED,
      "currency": null,
      "rate": null
    }
  },
```

**Comment by kroitor** [comment]
@ajoseps a quick question: do you place the market buy order with `createOrder`?

**Comment by ajoseps** [comment]
i'm not sure how the order was placed to be honest, it might have been on exchange.

**Comment by kroitor** [comment]
@ajoseps I've added a fix for it, so it should be ok as of 1.18.1158 (and it will be available for downloading in 10-15 minutes). However, the output you've posted above looks a bit weird, because the fee currency should've been properly detected, if the market was recognized. Can you confirm that `fee['currency'] === null` as seen on your side?

**Comment by ajoseps** [comment]
Confirmed that it is indeed `null`, this seems to only affect closed orders. With a quick sampling of closed orders it seems like they all have the fee currency set to `null`. Open orders seems fine. Thank you for the quick turnaround.
